include/roaring/roaring.h

```diff
@@ -879,25 +879,22 @@ Obviously, if you modify the underlying bitmap, the iterator
 becomes invalid. So don't.
 */
 
+/**
+ * A struct used to keep iterator state. Users should only access
+ * `current_value` and `has_value`, the rest of the type should be treated as
+ * opaque.
+ */
 typedef struct roaring_uint32_iterator_s {
-    const roaring_bitmap_t *parent;  // owner
-    int32_t container_index;         // point to the current container index
-    int32_t in_container_index;  // for bitset and array container, this is out
-                                 // index
-    int32_t run_index;           // for run container, this points  at the run
+    const roaring_bitmap_t *parent;  // Owner
+    int32_t container_index;         // Current container index
 
     uint32_t current_value;
     bool has_value;
 
-    const ROARING_CONTAINER_T
-        *container;  // should be:
-                     // parent->high_low_container.containers[container_index];
-    uint8_t typecode;  // should be:
-                       // parent->high_low_container.typecodes[container_index];
-    uint32_t highbits;  // should be:
-                        // parent->high_low_container.keys[container_index]) <<
-                        // 16;
-
+    const ROARING_CONTAINER_T *container;  // Current container
+    uint8_t typecode;                      // Typecode of current container
+    uint32_t highbits;                     // High 16 bits of the current value
+    roaring_container_iterator_t container_it;
 } roaring_uint32_iterator_t;
```

Comment:
@lemire, not sure what the expectation for CRoaring is for ABI compatibility, but this will break that, e.g. code like:

```c
roaring_uint32_iterator_t it;
roaring_init_iterator(bitmap, &it);
if (it.has_value) {
  //
}
```

will be totally broken if the version of the roaring header and the library don't match (the offset of `has_value` field has changed).

Reply:
Good point. If we need ABI compatibility it is possible to keep the struct as-is.

Reply:
> Roaring is for ABI compatibility,

I am currently planning on a major release so I think breaking ABI compatibility is acceptable.